theseus_gui/src/pages/Settings.vue

```diff
@@ -1,7 +1,471 @@
-<script setup></script>
+<script setup>
+import { ref, onBeforeMount } from 'vue'
+import {
+  Card,
+  Slider,
+  DropdownSelect,
+  Button,
+  SearchIcon,
+  PlayIcon,
+  Modal,
+  CheckIcon,
+  PlusIcon,
+  SaveIcon,
+  AnimatedLogo,
+} from 'omorphia'
+import { BrowseIcon } from '@/assets/icons'
+import { useTheming } from '@/store/state'
+import { get, set } from '@/helpers/settings'
+import { find_jre_8_jres, find_jre_17_jres } from '@/helpers/jre'
+import { open } from '@tauri-apps/api/dialog'
+
+const themeStore = useTheming()
+
+const loading = ref(false)
+const settings = ref({})
+const java8InstallOptions = ref([])
+const java17InstallOptions = ref([])
+
+onBeforeMount(async () => {
+  loading.value = true
+  settings.value = await get()
+
+  // Finding possible Java 8 and 17 installations
+  java8InstallOptions.value = await find_jre_8_jres()
+  java17InstallOptions.value = await find_jre_17_jres()
+
+  // Setting java version defaults. These can come as NULL from Tauri.
+  if (!settings.value.java_globals?.JAVA_8)
+    settings.value.java_globals.JAVA_8 = { path: '', version: '' }
+  if (!settings.value.java_globals?.JAVA_17)
+    settings.value.java_globals.JAVA_17 = { path: '', version: '' }
+  loading.value = false
+})
+
+// DOM refs
+const detectJava17Modal = ref(null)
+const detectJava8Modal = ref(null)
+
+const handleTheme = async (e) => {
+  themeStore.setThemeState(e.option.toLowerCase())
+  settings.value.theme = themeStore.selectedTheme
+  await set(settings.value)
+}
+const saveJavaPath = () => {}
+const saveSettings = async () => await set(settings.value)
+
+const handleJava17FileInput = async () => {
+  let filePath = await open()
+  settings.value.java_globals.JAVA_17 = {
+    path: filePath,
+    version: '17',
+  }
+}
+const handleJava8FileInput = async () => {
+  let filePath = await open()
+  settings.value.java_globals.JAVA_8 = {
+    path: filePath,
+    version: '8',
+  }
+}
+
+const setJava17Install = (chosenInstall) => {
+  settings.value.java_globals.JAVA_17 = chosenInstall
```

Comment:
This a weird bug, but this seems to be making it reactive to chosenInstall. If I select a Java version, then delete the string it put in the textbox, and then go back, the path disappears in the modal, and then I can't select it